### PR TITLE
Adding intel itt api

### PR DESCRIPTION
**Pull request by ZbigniewRA**
Specify library name and version:  **ittapi/3.23.0**

Intel ITT API (Instrumentation and Tracing Technology API) enables generation and control over profiling using Intel's VTune Profiler. This is a key API required to collect useful trace data for performance optimization work.
This API is similar in functionality to NVTX API, which targets Nvidia's Nsight Systems Profiler (`nvtx` already exists in Conan Center).

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ittapi/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
+import os
+
+required_conan_version = ">=1.47.0"
+
+
+class IttApiConan(ConanFile):
+    name = "ittapi"
+    license = "dual licensed under GPLv2 and 3-Clause BSD licenses"
```
Comment (uilianries) [inline recipes/ittapi/all/conanfile.py:11]:
```suggestion
    license = ("BSD-3-Clause", "GPL-2.0-only")
```
Only SPDX short names should be used, so we can filter 
https://spdx.org/licenses/

File: recipes/ittapi/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    def layout(self):
+        cmake_layout(self)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not cross_building(self):
```
Comment (uilianries) [inline recipes/ittapi/all/test_package/conanfile.py:23]:
```suggestion
        if can_run(self):
```
https://docs.conan.io/en/latest/reference/conanfile/tools/build.html#conan-tools-build-can-run

File: recipes/ittapi/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from conan import ConanFile
+from conan.tools.build import cross_building
```
Comment (uilianries) [inline recipes/ittapi/all/test_package/conanfile.py:2]:
```suggestion
from conan.tools.build import can_run
```

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. Please, add a test_v1_package folder, to validate COnan 1.x generators, as you are using Conan 2.x generators in your recipe. As template, you can use https://github.com/conan-io/conan-center-index/tree/master/docs/package/all/test_v1_package

File: recipes/ittapi/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        # We have no C++ files.
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
```
[the remark this answers is not in the document]
Comment (ZbigniewRA) [reply recipes/ittapi/all/conanfile.py:38]: I will make this change.
I'd like to point out though that the original code builds on MSVC successfully, so I'm confused.

File: recipes/ittapi/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    description = (
+        "The Instrumentation and Tracing Technology (ITT) API enables your application"
+        " to generate and control the collection of trace data during its execution"
+        " across different Intel tools."
+    )
+    topics = ("itt", "ittapi", "vtune", "profiler", "profiling")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],  # @note fPIC is always enabled by the underlying CMake file. So we ignore this option.
```
[the remark this answers is not in the document]
Comment (ZbigniewRA) [reply recipes/ittapi/all/conanfile.py]: Fixed.

**Comment by ZbigniewRA** [comment]
`test_v1_package` added as requested.

File: recipes/ittapi/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES CXX)
+
+find_package(ittapi REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE ittapi::ittapi)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```
Comment (prince-chrismc) [inline recipes/ittapi/all/test_package/CMakeLists.txt:8]: this should be in the recipe in the validate method 

https://github.com/conan-io/conan-center-index/blob/e7c65968be736eb9e46dd1c488a3b1d783f1b923/recipes/tuplet/all/conanfile.py#L38-L40

**Review by prince-chrismc — COMMENTED** [review]
One minor comment but it's super close

File: recipes/ittapi/all/test_package/CMakeLists.txt (hunk 6 above)
Comment (ZbigniewRA) [reply recipes/ittapi/all/test_package/CMakeLists.txt]: Thanks. Actually requirement for C++ 11 was not correct. I have removed it from test files.

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**